table/block_based/block_based_table_builder.cc

```diff
@@ -365,6 +365,9 @@ struct BlockBasedTableBuilder::Rep {
 
   std::unique_ptr<ParallelCompressionRep> pc_rep;
 
+  uint64_t get_offset() { return offset.load(std::memory_order_relaxed); }
+  void set_offset(uint64_t o) { offset.store(o, std::memory_order_relaxed); }
```

Comment:
For `r->offset`, I think it's a false positive from TSAN. When we're building data blocks, `r->offset` will only be set in writer. When we're appending meta blocks, parallel compression code path is disabled, so it will only be set in main thread.